platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/threading/ControllableQueue.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.test.threading;
+
+import com.swirlds.common.threading.framework.internal.AbstractBlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * This queue implementation allows us to artificially cause poll() to block.
+ */
+public class ControllableQueue extends AbstractBlockingQueue<Integer> {
+    private final ReentrantLock pollLock = new ReentrantLock();
+    private final AtomicInteger pollBlockedCount = new AtomicInteger(0);
+
+    public ControllableQueue() {
+        super(new LinkedBlockingQueue<>());
+    }
+
+    @Override
+    public synchronized Integer poll(long timeout, TimeUnit unit) throws InterruptedException {
+        pollBlockedCount.incrementAndGet();
+        pollLock.lock();
+        pollLock.unlock();
```

Comment:
I do not get the code / functionality of that class. Why do you lock a resource when you directly unlock it?